library/std/src/sys/unix/process/process_unix/tests.rs

```diff
@@ -0,0 +1,29 @@
+#[test]
+fn exitstatus_display_tests() {
+    // In practice this is the same on every Unix.
+    // If some weird platform turns out to be different, and this test fails, use #[cfg].
+    use crate::os::unix::process::ExitStatusExt;
+    use crate::process::ExitStatus;
+
+    let t = |v, s| assert_eq!(s, format!("{}", <ExitStatus as ExitStatusExt>::from_raw(v)));
+
+    t(0x0000f, "signal: 15");
+    t(0x0008b, "signal: 11 (core dumped)");
+    t(0x00000, "exit code: 0");
+    t(0x0ff00, "exit code: 255");
+
+    // On MacOS, 0x0137f is WIFCONTINUED, not WIFSTOPPED.  Probably *BSD is similar.
+    //   https://github.com/rust-lang/rust/pull/82749#issuecomment-790525956
+    // The purpose of this test is to test our string formatting, not our understanding of the wait
+    // status magic numbers.  So restrict these to Linux.
+    #[cfg(target_os = "linux")]
+    t(0x0137f, "stopped (not terminated) by signal: 19");
+    #[cfg(target_os = "linux")]
+    t(0x0ffff, "continued (WIFCONTINUED)");
```

Comment:
For tests that we only want to *run* on some platforms, but are capable of *compiling* perfectly fine on any platform, I would prefer to do the following instead. This makes things easier for someone who needs to refactor this code on a non-Linux host later. Completely conditional-compiling out the code would hide type errors or missing arguments and it commonly leads to needless round-trips through bors to get things right, which is a frustrating experience.

Using #[cfg] would typically only be if the compilation itself is platform-specific such as one of the std::os extension traits, or if for some reason there is immense cost in compiling the tests (which is not the case here).

```suggestion
    if cfg!(target_os = "linux") {
        t(0x0137f, "stopped (not terminated) by signal: 19");
        t(0x0ffff, "continued (WIFCONTINUED)");
    }
```

Reply:
> For tests that we only want to _run_ on some platforms, but are capable of _compiling_ perfectly fine on any platform, I would prefer to do the following instead. This makes things easier for someone who needs to refactor this code on a non-Linux host later. Completely conditional-compiling out the code would hide type errors or missing arguments and it commonly leads to needless round-trips through bors to get things right, which is a frustrating experience.

Thanks.  That is a very good point.